dev_guide/expanding_persistent_volumes.adoc

```diff
@@ -0,0 +1,39 @@
+[[expanding_persistent_volumes]]
+= Expanding Persistent Volumes
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+[[resizing-pv]]
+== Resizing Persistent Volumes
+
+You can expand persistent volume claims (PVC) online from {product-tile} for
+Containerized GlusterFS, Cinder, and Google Compute Engine (GCE) Persistent Disks (PD) by creating a storage class with
+`AllowVolumeExpansion=true`. The PVC uses the storage class and submits a claim.
+The PVC specifies a new increased size, and the underlying persistent volume
+(PV) is resized.
+
+[[e2e-online-expansion]]
+== End-to-end Online Expansion and Resize for Containerized GlusterFS Persistent Volumes
```

Comment:
IMO - there should be three distinct heading in this topic.

**Enabling Expansion of Persistent Volume Claims**
Volume expansion is a tech. preview feature and hence is not enabled by default in your {product-title} 3.9 cluster.  There may be other reasons because of which {product-title} admin may want to enable this feature only for certain use cases.  

To allow expansion of Persistent volume claims by {product-title} users, {product-title} admin must create or update a StorageClass with `allowVolumeExpansion` set to true. Only PVCs created from that class will be allowed expansion. 

Apart from that - {product-title} admin must enable `ExpandPersistentVolumes` feature flag and turn on `PersistentVolumeClaimResize` admission controller. Please refer to <link_here> section for how to enable feature flag and how to enable a admission controller.

**Expanding GlusterFS based Persistent Volume Claims**
Expanding GlusterFS volumes is easiest. Once {product-title} admin has created a storage class with `allowVolumeExpansion` set to true, you can create a Persistent Volume Claim from that class and afterwards whenever needed you can just edit PVC and request new size:

For example given:

```yaml
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: gluster-mysql
spec:
  storageClass: "storageClassWithFlagSet"
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 8Gi
```

You can request expanded volume by just updating `spec.resources.requests` 

**Expanding Persistent Volume Claims with File System**
For expanding PVCs based off volume types that need File system resizing(such as  - GCE-PD, EBS, Cinder) is a two step process. Usually it involves - expanding volume object in CloudProvider and then expanding file system on actual node.

Expanding file system on node only happens when a new pod is started with the volume. 

- Step 1: Assuming PVC was created from a storageClass with `allowVolumeExpansion` set to `true`. You can go ahead and edit the PVC and request new size by editing `spec.resources.requests`.
- Step 2: Once CloudProvider object has finished resizing. The PVC will have condition `FileSystemResizePending` set. You can run following command to check the Condition:

```
~> oc describe pvc <pvc_name>
```

When CloudProvider object has finished resizing, the Persistent Volume object (PV) will reflect newly requested size in `PersistentVolume.Spec.Capacity` .  At this point, you can create a create or re-create new Pod from PVC to finish File system resizing. Once pod is running - it should have newly requested size available and `FileSystemResizePending` condition should be removed from PVC.